### DISCUSSION TITLE
Charge Period Card for H3

### DISCUSSION THREAD

**Discussion by Kasttolo** [nathanmarlor/foxess_modbus_charge_period_card#12] [Q&A] [answered]
Hi, 
I am using the Modbus integration for a few month and I am really happy with it, many thanks for the fantastic work on it.
The Carge period card is unfortunately not working for me, as I am having an H3-E. Are there any plans to expand the use of the carge period cards to the H3? Best Regards

**Comment by canton7** [comment, marked as answer]
The H3 doesn't support setting charge periods over modbus unfortunately

**Reply by Kasttolo** [reply]
That's a pitty, but thanks for your reply. Hopefully it changes with some future Firmware updates

**Reply by Gics78** [reply]
Now that the H3 does - can you implement it? Right now the card is stuck "loading"

**Reply by canton7** [reply]
The H3 does not support configuring the charge periods over modbus.

It is possible to tell it to force charge and force discharge, which means you can create automations to charge/discharge on a schedule. However it is not possible to configure the two charge periods, the same way you can through the app, and therefore it is not possible for this charge period card to work with the H3.

**Reply by Gics78** [reply]
Got it, thanks! I messed "charge periods" up with "force charge". Well - the result should be the same
